### PR TITLE
In Progress Implmentation for SkillsBasedRouting modification

### PR DESCRIPTION
**Pull request by CRMTravis**

### CODE REVIEW

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 1]
```diff
@@ -142,26 +189,36 @@ global class SkillsBasedRouting {
...
-          new Worker(record.RelatedRecordId, 0, skillRecord.Skill.MasterLabel)
-        );
+        if (locations.contains(skillRecord.Skill.MasterLabel)) {
+          serviceResourceLoactionSkillNames.add(skillRecord.Skill.MasterLabel);
+        } else if (products.contains(skillRecord.Skill.MasterLabel)) {
+          serviceResourceProductSkillNames.add(skillRecord.Skill.MasterLabel);
+        }
+      }
+      for (String location : serviceResourceLoactionSkillNames) {
+        for (String product : serviceResourceProductSkillNames) {
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:199]: serviceResourceLoactionSkillNames and serviceResourceProductSkillNames need to be related by their UserId to work, consider multiple users.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/ServiceResourceWrapper.cls [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * @description       :
+ * @author            : Travis Allen
+ * @group             :
+ * @last modified on  : 04-21-2021
+ * @last modified by  : Travis Allen
+ * Modifications Log
+ * Ver   Date         Author         Modification
+ * 1.0   04-21-2021   Travis Allen   Initial Version
+ **/
+public with sharing class ServiceResourceWrapper {
```
Comment (RemarioCRM) [inline force-app/main/default/classes/ServiceResourceWrapper.cls:11]: This should be an inner class

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/ServiceResourceWrapper.cls (hunk 2 above)
Comment (CRMTravis) [reply force-app/main/default/classes/ServiceResourceWrapper.cls:11]: Why exactly?

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/TestSkillsBasedRouting.cls [hunk 3]
```diff
@@ -115,24 +115,32 @@ public class TestSkillsBasedRouting {
...
 
-    ServiceResourceSkill abigailCreditCardSkill = TestDataFactory.CreateServiceResourceSkill(
+    ServiceResourceSkill abigailBarbicanSkill = TestDataFactory.CreateServiceResourceSkill(
       abigailSR.Id,
-      creditCardSkill.Id
+      barbicanSkill.Id
     );
-    serviceResourceSkills.add(abigailCreditCardSkill);
+    serviceResourceSkills.add(abigailBarbicanSkill);
+
```
Comment (RemarioCRM) [inline force-app/main/default/classes/TestSkillsBasedRouting.cls:129]: Remove all dead commented code

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 4]
```diff
@@ -183,72 +114,44 @@ global class SkillsBasedRouting {
...
-      );
-
       for (ServiceResourceSkill skillRecord : record.ServiceResourceSkills) {
-        System.debug(TAG + ' Name Of Skill ' + skillRecord.Skill.MasterLabel);
-        if (Constants.SKILL_LOCATIONS.contains(skillRecord.Skill.MasterLabel)) {
+        if (Constants.SKILL_PRODUCTS.contains(skillRecord.Skill.MasterLabel)) {
           if (serviceResourceMap.keySet().contains(record.RelatedRecordId)) {
             serviceResourceMap.get(record.RelatedRecordId)
-              .addLocation(skillRecord.Skill.MasterLabel);
+              .addProduct(skillRecord.Skill.MasterLabel);
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:121]: For all the grouped queries , ADD with security enforced

Comment (RemarioCRM) [reply force-app/main/default/classes/SkillsBasedRouting.cls:121]: ensure the claass uses with sharing

File: force-app/main/default/classes/TestSkillsBasedRouting.cls [hunk 5]
```diff
@@ -222,27 +237,97 @@ public class TestSkillsBasedRouting {
       ];
 
       List<Lead> newLead = TestDataFactory.CreateLeads(1);
-      newLead[0].Preferred_Location__c = 'Anotto Bay';
-      newLead[0].Service_of_Interest__c = 'JN Bank AutoLoan';
+      newlead[0].LastName = 'Travis';
+      newLead[0].Preferred_Location__c = 'Annotto Bay';
+      newLead[0].Service_of_Interest__c = 'JN Bank Auto Loan';
       newLead[0].Escalate_to_Tier2__c = false;
       newLead[0].Last_Contact_Made__c = System.now();
```
Comment (RemarioCRM) [inline force-app/main/default/classes/TestSkillsBasedRouting.cls:244]: great job, but im not seeing no negative test scenario when the user does not the matching location or product.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

**Review by RemarioCRM — APPROVED** [review]

**Pull request merged**